src/cartridge.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    fs::File,
+    io::{self, Read},
+    rc::Rc,
+};
+
+///Contains all the information about the header of the cartridge
+pub(crate) struct CartridgeHeader {
```

Comment:
The comment here doesn't add relevant information, is better to explain what is it or the behavior or other important thing, you are saying just is a thing that contains information.

I my opinion is better to keep in separated file just because if we will add tests it will be very long to read, but let's see what other members think.